### PR TITLE
CI: Remove warning raising after new matplotlib release

### LINKED ISSUE #31564: CI: deprecation warning tests failing on linux py37_local pipeline
#### Problem description

As of now, the Linux py37_locale pipeline seems to fail independent of PR:

```
_________________ TestRegistration.test_registering_no_warning _________________
2020-02-02T08:55:26.6331296Z [gw1] linux -- Python 3.7.6 /home/vsts/miniconda3/envs/pandas-dev/bin/python

self = <pandas.tests.plotting.test_converter.TestRegistration object at 0x7f97699062d0>

     def test_registering_no_warning(self):
         plt = pytest.importorskip("matplotlib.pyplot")
         s = Series(range(12), index=date_range("2017", periods=12))
         _, ax = plt.subplots()
     
        # Set to the "warn" state, in case this isn't the first test run
        register_matplotlib_converters()
        with tm.assert_produces_warning(DeprecationWarning, check_stacklevel=False):
            # GH#30588 DeprecationWarning from 2D indexing
>            ax.plot(s.index, s.values)
E               AssertionError: Did not see expected warning of class 'DeprecationWarning'
```

#### Full failures output

<details>

```
2020-02-02T08:55:26.6329275Z =================================== FAILURES ===================================
2020-02-02T08:55:26.6329476Z _________________ TestRegistration.test_registering_no_warning _________________
2020-02-02T08:55:26.6331296Z [gw1] linux -- Python 3.7.6 /home/vsts/miniconda3/envs/pandas-dev/bin/python
2020-02-02T08:55:26.6331555Z 
2020-02-02T08:55:26.6331736Z self = <pandas.tests.plotting.test_converter.TestRegistration object at 0x7f97699062d0>
2020-02-02T08:55:26.6331919Z 
2020-02-02T08:55:26.6332094Z     def test_registering_no_warning(self):
2020-02-02T08:55:26.6332272Z         plt = pytest.importorskip("matplotlib.pyplot")
2020-02-02T08:55:26.6332458Z         s = Series(range(12), index=date_range("2017", periods=12))
2020-02-02T08:55:26.6332629Z         _, ax = plt.subplots()
2020-02-02T08:55:26.6332795Z     
2020-02-02T08:55:26.6333232Z         # Set to the "warn" state, in case this isn't the first test run
2020-02-02T08:55:26.6333431Z         register_matplotlib_converters()
2020-02-02T08:55:26.6333618Z         with tm.assert_produces_warning(DeprecationWarning, check_stacklevel=False):
2020-02-02T08:55:26.6333793Z             # GH#30588 DeprecationWarning from 2D indexing
2020-02-02T08:55:26.6333984Z >           ax.plot(s.index, s.values)
2020-02-02T08:55:26.6334129Z 
2020-02-02T08:55:26.6334293Z pandas/tests/plotting/test_converter.py:71: 
2020-02-02T08:55:26.6334479Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2020-02-02T08:55:26.6334637Z 
2020-02-02T08:55:26.6334800Z self = <contextlib._GeneratorContextManager object at 0x7f9769b95250>
2020-02-02T08:55:26.6334986Z type = None, value = None, traceback = None
2020-02-02T08:55:26.6335129Z 
2020-02-02T08:55:26.6335310Z     def __exit__(self, type, value, traceback):
2020-02-02T08:55:26.6335477Z         if type is None:
2020-02-02T08:55:26.6335637Z             try:
2020-02-02T08:55:26.6335813Z >               next(self.gen)
2020-02-02T08:55:26.6336232Z E               AssertionError: Did not see expected warning of class 'DeprecationWarning'
2020-02-02T08:55:26.6336444Z 
2020-02-02T08:55:26.6336863Z ../../../miniconda3/envs/pandas-dev/lib/python3.7/contextlib.py:119: AssertionError
2020-02-02T08:55:26.6337076Z ___________________ TestRegistration.test_option_no_warning ____________________
2020-02-02T08:55:26.6337500Z [gw1] linux -- Python 3.7.6 /home/vsts/miniconda3/envs/pandas-dev/bin/python
2020-02-02T08:55:26.6337681Z 
2020-02-02T08:55:26.6337867Z self = <pandas.tests.plotting.test_converter.TestRegistration object at 0x7f9769c690d0>
2020-02-02T08:55:26.6338027Z 
2020-02-02T08:55:26.6338188Z     def test_option_no_warning(self):
2020-02-02T08:55:26.6338368Z         pytest.importorskip("matplotlib.pyplot")
2020-02-02T08:55:26.6338539Z         ctx = cf.option_context("plotting.matplotlib.register_converters", False)
2020-02-02T08:55:26.6338733Z         plt = pytest.importorskip("matplotlib.pyplot")
2020-02-02T08:55:26.6338903Z         s = Series(range(12), index=date_range("2017", periods=12))
2020-02-02T08:55:26.6339083Z         _, ax = plt.subplots()
2020-02-02T08:55:26.6339242Z     
2020-02-02T08:55:26.6339402Z         # Test without registering first, no warning
2020-02-02T08:55:26.6339586Z         with ctx:
2020-02-02T08:55:26.6339759Z             # GH#30588 DeprecationWarning from 2D indexing on Index
2020-02-02T08:55:26.6339947Z             with tm.assert_produces_warning(DeprecationWarning, check_stacklevel=False):
2020-02-02T08:55:26.6340121Z >               ax.plot(s.index, s.values)
2020-02-02T08:55:26.6340414Z 
2020-02-02T08:55:26.6340592Z pandas/tests/plotting/test_converter.py:105: 
2020-02-02T08:55:26.6340763Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2020-02-02T08:55:26.6340928Z 
2020-02-02T08:55:26.6341093Z self = <contextlib._GeneratorContextManager object at 0x7f976ac1aa90>
2020-02-02T08:55:26.6341279Z type = None, value = None, traceback = None
2020-02-02T08:55:26.6341440Z 
2020-02-02T08:55:26.6341601Z     def __exit__(self, type, value, traceback):
2020-02-02T08:55:26.6341764Z         if type is None:
2020-02-02T08:55:26.6341941Z             try:
2020-02-02T08:55:26.6342099Z >               next(self.gen)
2020-02-02T08:55:26.6342547Z E               AssertionError: Did not see expected warning of class 'DeprecationWarning'
2020-02-02T08:55:26.6342814Z 
2020-02-02T08:55:26.6343228Z ../../../miniconda3/envs/pandas-dev/lib/python3.7/contextlib.py:119: AssertionError
```

</details>

So this in two PRs that have gone through checks recently, and none of them touch anything that relates to the failing tests (#31562 #31559).

### PR DESCRIPTION
**Pull request by charlesdong1991**
- [ ] xref #31562   
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

seems CI is broken after the new release of matplotlib

### CODE REVIEW

**Comment by charlesdong1991** [comment]
xref #30588

File: pandas/tests/plotting/test_converter.py [hunk 1]
```diff
@@ -66,9 +66,16 @@ def test_registering_no_warning(self):
 
         # Set to the "warn" state, in case this isn't the first test run
         register_matplotlib_converters()
-        with tm.assert_produces_warning(DeprecationWarning, check_stacklevel=False):
-            # GH#30588 DeprecationWarning from 2D indexing
+
+        import matplotlib as mpl
+
+        version = mpl.__version__
+        if version > "3.1.2":
```
Comment (jreback) [inline pandas/tests/plotting/test_converter.py]: use LooseVersion

Comment (jreback) [reply pandas/tests/plotting/test_converter.py]: can also just define a new mpl skip decorator and skip if < 3.1.2

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/plotting/test_converter.py (hunk 1 above)
Comment (charlesdong1991) [reply pandas/tests/plotting/test_converter.py]: thanks for your quick reply, changed, and let me know if this is what you meant @jreback

File: pandas/tests/plotting/test_converter.py [hunk 2]
```diff
@@ -59,16 +60,15 @@ def test_register_by_default(self):
         call = [sys.executable, "-c", code]
         assert subprocess.check_call(call) == 0
 
+    @td.skip_if_low_mpl
```
Comment (jreback) [inline pandas/tests/plotting/test_converter.py]: yes right idea 

i think we have some skips that are based on mpl versions already (maybe in the other test plotting files)

can u name like them

Comment (charlesdong1991) [reply pandas/tests/plotting/test_converter.py]: emm, i see we have `skip_if_no_mpl` and `skip_if_mpl` decorators, but seems no version involved.

Comment (charlesdong1991) [reply pandas/tests/plotting/test_converter.py]: does `skip_if_old_mpl` sound better? or `skip_if_lower_3_1_2_mpl`?

Comment (jreback) [reply pandas/tests/plotting/test_converter.py]: latter

File: pandas/tests/plotting/test_converter.py [hunk 3]
```diff
@@ -91,6 +91,7 @@ def test_matplotlib_formatters(self):
                 assert Timestamp not in units.registry
             assert Timestamp in units.registry
 
+    @td.skip_if_lower_3_1_2_mpl
```
Comment (jreback) [inline pandas/tests/plotting/test_converter.py]: sorry we don't really want to add this as a full decorator, just do this

``@td.skip_if_no("matplotlib", min_version="3.1.2")```

**Review by jreback — CHANGES_REQUESTED** [review]
comments, ping on green.

**Review by jreback — APPROVED** [review]
great. ping on green.

**Comment by charlesdong1991** [comment]
this looks better simpler, i didn't notice this `skip_if_no` decorator, thanks for the tip! @jreback 

since it uses `<=` in `safe_import`, so setting the `min_version` to `3.1.3`, it works locally for me

**Comment by charlesdong1991** [comment]
ping @jreback

**Comment by jreback** [comment]
thanks for pro-actively fixing this @charlesdong1991 !

**Pull request merged**